cmd/anycastmonitor/main.go

```diff
@@ -135,3 +153,40 @@ func (j *anycastMonitorJob) ValidateImages(manifestRef keppel.ManifestReference)
 		}
 	}
 }
+
+func (j *anycastMonitorJob) ValidateAnycastMembership(apiPublicHostname string) {
+	resp, err := http.Get("https://keppel.global.cloud.sap/keppel/v1/auth?service=keppel.global.cloud.sap&scope=repository:foo/bar:pull")
+	if err != nil {
+		logg.Error("failed getting anon token: %s", err.Error())
+	}
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		logg.Error("failed reading body: %s", err.Error())
+	}
+
+	var data authapi.TokenResponse
+	err = json.Unmarshal(body, &data)
+	if err != nil {
+		logg.Error("failed to unmarshal JWT: %s", err.Error())
+	}
+	token, err := base64.StdEncoding.DecodeString(strings.Split(data.Token, ".")[1])
```

Comment:
You should check the length of the slice returned from `strings.Split` before proceeding, to avoid a crash if `data.Token` is malformed and contains no dots at all.